lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/type/DecodedTimestamp.java

```diff
@@ -13,15 +13,6 @@
  */
 package io.trino.plugin.base.type;
 
-import static com.google.common.base.Preconditions.checkArgument;
-import static java.util.concurrent.TimeUnit.SECONDS;
-
 public record DecodedTimestamp(long epochSeconds, int nanosOfSecond)
 {
-    private static final long NANOS_PER_SECOND = SECONDS.toNanos(1);
-
-    public DecodedTimestamp
-    {
-        checkArgument(nanosOfSecond >= 0 && nanosOfSecond < NANOS_PER_SECOND, "Invalid value for nanosOfSecond: %s", nanosOfSecond);
```

Comment:
If we need to ignore the nanosOfSecond for parquet data because third-party tools are broken, then we should contain it to the parquet reader. However, we do want to keep this validation for everything else.

Doing so will require "fixing" the broken data on read in the parquet reader.